lib/taskana-core/src/main/java/pro/taskana/mappings/QueryMapper.java

```diff
@@ -32,7 +32,14 @@
         + "<where>"
         + "<if test='accessIdIn != null'> "
         + "AND t.WORKBASKET_ID IN ( "
+        + "<choose>"
+        + "<when test=\"_databaseId == 'db2'\">"
```

Comment:
The explicit cast from bool -> int was introduced for the postgresql and is only necessary for the postgresql. Instead of not casting only for the db2 instance i'd rather only cast when a postgres db is found.
This does not change the outcome, but I think that it will make it a little clearer why we decided to handle it this way. [ Furthermore the extra processing (even if it's very little) will only be done from the postgresql and not from the other databases. ]

Reply:
we should make the more unusual case the special case. in this case db2, because boolean is supported by most databases and will be in DB2 11, won't it? This way we can remove the special case easier later on...

Reply:
Yes, all our other dbs support the (::) cast. I just did a quick research and didn't figure out which dbs support that cast. As far as I remember i'd say that most of them do (don't take my word on this one)   

So yes, we can leave it as it is in order to remove it easier.